vizier/backend/src/info/vizierdb/commands/plot/BarChart.scala

```diff
@@ -0,0 +1,164 @@
+import info.vizierdb.artifacts.VegaFrom
+import info.vizierdb.artifacts.VegaChart
+import info.vizierdb.artifacts.VegaScale
+import info.vizierdb.artifacts.VegaScaleType
+import info.vizierdb.artifacts.VegaAxis
+import info.vizierdb.artifacts.VegaOrientation
+import info.vizierdb.artifacts.VegaMarkEncoding
+import info.vizierdb.artifacts.VegaMarkEncodingGroup
+import info.vizierdb.artifacts.VegaDomain
+import info.vizierdb.artifacts.VegaRange
+import info.vizierdb.artifacts.VegaAutosize
+import info.vizierdb.artifacts.VegaPadding
+import info.vizierdb.artifacts.VegaLegend
+import info.vizierdb.artifacts.VegaLegendType
+
+object BarChart extends Command
+{
+  val PARAM_SERIES = "series" 
+  val PARAM_DATASET = "dataset"
+  val PARAM_X = "xcol"
+  val PARAM_Y = "ycol"
+  val PARAM_FILTER = "filter"
+  val PARAM_COLOR = "color"
+  val PARAM_LABEL = "label"
+  val PARAM_ARTIFACT = "artifact"
+  val PARAM_SORT = "sort"
+
+  override def name: String = "Bar Chart"
+
+  override def parameters: Seq[Parameter] = Seq(
+    ListParameter(id = PARAM_SERIES, name = "Bars", components = Seq(
+      DatasetParameter(id = PARAM_DATASET, name = "Dataset"),
+      ColIdParameter(id = PARAM_X, name = "X-axis"),
+      ColIdParameter(id = PARAM_Y, name = "Y-axis"),
+      StringParameter(id = PARAM_LABEL, name = "Label", required = false),
+      StringParameter(id = PARAM_FILTER, name = "Filter", required = false, helpText = Some("e.g., state = 'NY'")),
+      StringParameter(id = PARAM_COLOR, name = "Color", required = false, helpText = Some("e.g., #214478")),
+      BooleanParameter(id = PARAM_SORT, name = "Sort", required = true, default = Some(false))
+    )),
+    StringParameter(id = PARAM_ARTIFACT, name = "Output Artifact (blank to show only)", required = false)
+  )
+  override def title(arguments: Arguments): String = 
+    "Bar plot of "+arguments.getList(PARAM_SERIES).map { series =>
+      series.get[String](PARAM_DATASET)
+    }.toSet.mkString(", ")
+
+  override def format(arguments: Arguments): String =
+    title(arguments)
+
+  override def process(arguments: Arguments, context: ExecutionContext): Unit =
+  {
+    // Figure out if we are being asked to emit a named artifact
+    // Store the result in an option-type
+    val artifactName = arguments.getOpt[String](PARAM_ARTIFACT)
+                                .flatMap { case "" => None 
+                                           case x => Some(x) }
+
+    // Feed the configuration into PlotUtils
+    val series =
+      PlotUtils.SeriesList( 
+        arguments.getList(PARAM_SERIES).map { series => 
+          PlotUtils.makeSeries(
+            context     = context,
+            datasetName = series.get[String](PARAM_DATASET),
+            xIndex      = series.get[Int](PARAM_X),
+            yIndex      = series.get[Int](PARAM_Y),
+            filter      = series.getOpt[String](PARAM_FILTER),
+            sort        = series.get[Boolean](PARAM_SORT),
```

Comment:
add `name = series.get[String](PARAM_LABEL)`.  The lack of this in the other charts was a bug.